### ISSUE TITLE
False positive cannot move when str == str

### ISSUE THREAD

**Issue by 0xrgb** [intellij-rust/intellij-rust#3314]
## Environment

* **Intellij-Rust plugin version:** `0.2.91.2115-183`
* **Rust toolchain version:** `rustc 1.32.0 (stable-x86_64-pc-windows-gnu)`
* **IDE name and version:** `CLion 2018.3.3`
* **Operating system:** `Windows 10 (x64)`

## Problem description

IJ reports a "Cannot move" error when comparing two `str`, but it is legal upon compilation.

## Steps to reproduce

```rust
fn main() {
    let a = "abc";
    let b = "abc";

    // Cannot move less... (Ctrl+F1)
    // Inspection info: Implements simple cases of borrow checking. 
    if *a == *b {
        println!("Same str");
    }
}
```
